### PR TITLE
Misc updates

### PR DESCRIPTION
**Pull request by dhinakg**
TODO (doable after merge):
* Convert `download.developer.apple.com` links 
* Rewrite the build system to generate link variants on build
* Add metadata for local IPSWs

### CODE REVIEW

File: osFiles/macOS/20x - 11.x/20C5061b.json [hunk 1]
```diff
@@ -94,8 +94,6 @@
                 "MacBookAir8,2",
                 "MacBookAir9,1",
                 "MacBookAir10,1",
-                "MacBookPro11,1-Late-2013",
```
Comment (dhinakg) [inline osFiles/macOS/20x - 11.x/20C5061b.json:97]: Intentional?

File: osFiles/macOS/21x - 12.x/21F2092.json [hunk 2]
```diff
@@ -11,50 +11,95 @@
...
+                "sha1": "0c17176fb5d59a0eccb1db5811a7b5b6dbc6a1a3",
+                "sha2-256": "e81de9b2636d004d6e18dbc70bacfbd248850abcf3fad2e439f5c8a92c22fc12"
             },
-            "size": 11676595623
+            "size": 13697592054
         },
         {
-            "type": "ipsw",
+            "type": "installassistant",
             "deviceMap": [
```
Comment (dhinakg) [inline osFiles/macOS/21x - 12.x/21F2092.json:88]: this looks broken

File: osFiles/macOS/20x - 11.x/20C5061b.json (hunk 1 above)
Comment (ROSeaboyer) [reply osFiles/macOS/20x - 11.x/20C5061b.json:97]: Triple-checked, and this is around where Big Sur was bricking the 13" MBP

File: osFiles/macOS/21x - 12.x/21F2092.json (hunk 2 above)
Comment (ROSeaboyer) [reply osFiles/macOS/21x - 12.x/21F2092.json:43]: I think the order just got reversed; I did also fix up the root-level deviceMap just now as I noticed that had a full list

**Pull request merged**